vault/logical_system.go

```diff
@@ -388,7 +398,18 @@ func (b *SystemBackend) handlePluginCatalogDelete(ctx context.Context, req *logi
 	if pluginName == "" {
 		return logical.ErrorResponse("missing plugin name"), nil
 	}
-	pluginType, err := consts.ParsePluginType(d.Get("type").(string))
+
+	pluginTypeStr := d.Get("type").(string)
+	if pluginTypeStr == "" {
+		// If the plugin type is not provided (i.e. the old
+		// sys/plugins/catalog/:name endpoint is being requested) short-circuit here
+		// and return a warning
+		resp := &logical.Response{}
+		resp.AddWarning(fmt.Sprintf("Deprecated API endpoint, cannot deregister plugin from catalog for %q", pluginName))
+		return resp, nil
```

Comment:
I do think we should still attempt to delete here. We need an escape hatch incase we are unable to upgrade a plugin. Otherwise the plugin will never be updated and log on every startup

Reply:
So if pluginType == unknown then delete out of the old storage

Reply:
I think the pluginCatalog will do this already